### PR TITLE
RN Turbo/Fabric migration

### PR DESCRIPTION
**Pull request by ian-wd**
# Pull Request: react-native-sketch-canvas

## Description
Turbo/Fabric View migration

## Type of Change

- [ ] New Plugin
- [ ] Feature Addition
- [ ] Bug Fix
- [ ] Performance Improvement
- [x] Code Refactoring
- [ ] Documentation Update
- [ ] Other (please describe):

## Testing Performed

- [x] Unit Tests
- [ ] Integration Tests
- [x] Component Tests
- [x] Manual Testing
- [ ] Edge Cases Verified

## Test Results
image

## Breaking Changes
- Bumped minimum RN version to v0.76.7
- New Architecture must be enabled

### CODE REVIEW

**Comment by nemanjar7** [comment]
Screenshot 2025-03-12 at 1 39 56 PM

Are these logs supposed to be here?
Screenshot 2025-03-12 at 1 34 01 PM


Android:

https://github.com/user-attachments/assets/a8f29209-406c-4dd7-a1ff-a16fc2c8fb9f


https://github.com/user-attachments/assets/cfe74e01-6ade-4fd7-8494-c6932294e314


iOS:


https://github.com/user-attachments/assets/129a5795-8a1e-4483-b784-de30749086a2


https://github.com/user-attachments/assets/02fff070-1665-4016-bc6f-654f0d97c506


Seems fine to me, the only thing I noticed is that if we don't clear the screen on one canvas, it tends to transfer those changes to the new screen. Not sure if the app is set up that way since it's just example app.

**Comment by iBotPeaches** [comment]
The logs are intentional since just a sample app to see what's it's doing. The drawing persisting seems odd - I'd like @ian-wd to comment on that.

**Comment by ian-wd** [comment]
> The logs are intentional since just a sample app to see what's it's doing. The drawing persisting seems odd - I'd like @ian-wd to comment on that.

ill check this weekend, I haven't encountered that I'm mostly running this on android

**Comment by ian-wd** [comment]
Fixed iOS fabric view/sketchcanvas cleanup

https://github.com/user-attachments/assets/6d326180-098b-4f2b-81d0-99a64fb37e2e

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -12,26 +12,32 @@ jobs:
...
           registry-url: 'https://registry.npmjs.org'
 
-      - name: NPM Install
-        run: npm ci
+      - name: Yarn Install
+        run: yarn install --immutable
 
       - name: Publish (NPM)
-        run: npm publish --access public
+        run: yarn npm publish --access public
```
Comment (iBotPeaches) [inline .github/workflows/publish.yml:30]: Prefixing yarn in front of an npm command seems odd

**Comment by iBotPeaches** [comment]
@nemanjar7 - May you confirm your issues you identified with iOS clearing is good? I think this is ready to merge if so.

**Comment by nemanjar7** [comment]
Okay, few findings:
- The previous issues with screens not clearing are corrected
- I noticed some other issues, for examples 6 and 7 (Video was too long so I only updated the issues with ex6 and ex7). When you get in the example 6 first time it shows these different kinds of text, so I just scribbled something and closed the screen, but when you get back on it, it shows blank screen. Also, not in the video, but changing text at the top, modifies only text on the left side. This could be though me not fully understanding the functionality of this particular screen.
- Last screen has similar issue in a sense where returning to the screen after you just got out of it (and not checking any other screen) will show different UI.
- If you switch screens (examples) however, everything is reset properly besides example 7. That one is very inconsistent since sometimes it shows 2 canvases, sometimes 1, and then sometimes nothing.

https://github.com/user-attachments/assets/0beea77a-0e2f-4683-b3fc-d97c036ab79d


Also I want to include a possibility that I maybe didn't set up something right, since when I tried to run the app from `xcode` is got the following error below(no matter what I did, I couldn't make it work like that). However, after running it from `vscode` with `npm run ios` it launched with no issues.

Screenshot 2025-03-31 at 3 54 58 PM

**Comment by nemanjar7** [comment]
This is the other things I talked about.

https://github.com/user-attachments/assets/74824a81-760d-42f7-89cb-03d6e00c3527

**Comment by ian-wd** [comment]
> This is the other things I talked about. So if you have time to test it out maybe to confirm it's not just my local env? @iBotPeaches
> 
>  extra.mov

yeah this looks like regression @nemanjar7, probably related to the view cleanup

**Comment by nemanjar7** [comment]
I believe it's g2g now. @iBotPeaches 

https://github.com/user-attachments/assets/cf708e1b-5211-450e-b855-f84f589e0a93

**Review by nemanjar7 — DISMISSED** [review]

**Review by iBotPeaches — APPROVED** [review]
